bootstrap/kubeadm/api/v1alpha3/condition_consts.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+
+// Conditions and condition Reasons for the KubeadmConfig object
+
+const (
+	// DataSecretAvailableCondition documents the status of the bootstrap secret generation process.
+	//
+	// NOTE: When the DataSecret generation starts the process completes immediately and within the
+	// same reconciliation, so the user will always see a transition from Wait to Generated without having
+	// evidence that BootstrapSecret generation is started/in progress.
+	DataSecretAvailableCondition clusterv1.ConditionType = "DataSecretAvailable"
+
+	// WaitingForClusterInfrastructureReason (Severity=Info) document a bootstrap secret generation process
+	// waiting for the cluster infrastructure to be ready.
+	//
+	// NOTE: Having the cluster infrastructure ready is a pre-condition for starting to create machines;
+	// the KubeadmConfig controller ensure this pre-condition is satisfied.
+	WaitingForClusterInfrastructureReason = "WaitingForClusterInfrastructure"
+
+	// WaitingForControlPlaneInitializedReason (Severity=Info) document a bootstrap secret generation process
+	// waiting for the first control plane machine to be ready.
+	//
+	// NOTE: Having the first control plane machine ready is a pre-condition for the additional control planes and
+	// for workers to join; the KubeadmConfig controller ensure this pre-condition is satisfied.
+	WaitingForControlPlaneInitializedReason = "WaitingForControlPlaneInitialized"
```

Comment:
Is this same as control plane being ready?

Reply:
In order to unblock this, it is required that the first control plane instance is up and running.
For historical reason, we have both ready and initialized to track this (with slight differences) but given that for condition Ready can be confusing, I'm using Initialized